jobsdb/jobsdb.go

```diff
@@ -3007,7 +3003,28 @@ func (jd *HandleT) backupTable(backupDSRange dataSetRangeT, isJobStatusTable boo
 	return true, nil
 }
 
-func (jd *HandleT) getBackupDSRange() dataSetRangeT {
+func (jd *HandleT) backupUploadWithExponentialBackoff(ctx context.Context, file *os.File, pathPrefixes ...string) (filemanager.UploadOutput, error) {
+	// get a file uploader
+	fileUploader, errored := jd.getFileUploader()
+	jd.assertError(errored)
```

Comment:
If we return an error instead of panic, the operation will be done with just an error. Maybe we don't want this.

```go
			err := jd.backupDS(ctx, backupDSRange)
			if err != nil {
				jd.logger.Errorf("[JobsDB] :: Failed to backup jobs table %v. Err: %v", backupDSRange.ds.JobStatusTable, err)
			}
			jd.JournalMarkDone(opID)
```

Reply:
That should be okay right @lvrach? Instead of panicking.

Reply:
If we are not able to create the file manager, we will conclude the operation and print an error. Sounds good to me.

Reply:
I just copied this piece of code into exponentialBackoff function without giving much thought to it. But, yes it makes sense to not assert. Removed it.